src/robot/libdocpkg/htmlwriter.py

```diff
@@ -142,25 +142,58 @@ def _get_formatter(self, doc_format):
         try:
             return {'ROBOT': html_format,
                     'TEXT': self._format_text,
-                    'HTML': self._format_html,
+                    'HTML': lambda doc: doc,
                     'REST': self._format_rest}[doc_format]
         except KeyError:
             raise DataError("Invalid documentation format '%s'." % doc_format)
 
     def _format_text(self, doc):
         return '<p style="white-space: pre-wrap">%s</p>' % html_escape(doc)
 
-    def _format_html(self, doc):
-        return '<div style="margin: 0">%s</div>' % doc
-
     def _format_rest(self, doc):
         try:
             from docutils.core import publish_parts
         except ImportError:
             raise DataError("reST format requires 'docutils' module to be installed.")
         parts = publish_parts(doc, writer_name='html',
                               settings_overrides={'syntax_highlight': 'short'})
-        return self._format_html(parts['html_body'])
+        return parts['html_body']
 
     def __call__(self, doc):
         return self._formatter(doc)
+
+
+class HtmlToText(object):
+
+    html_tags = {
+        'b': '*',
+        'i': '_',
+        'strong': '*',
+        'em': '_',
+        'code': '``'
+    }
+
+    single_chars = {
+        '<br>': '\n',
+        '<br/>': '\n',
+        '&amp;': '&',
+        '&lt;': '<',
+        '&gt;': '>',
+        '&quot;': '"',
+        '&apos;': "'"
+    }
+
+    def get_shortdoc_from_html(self, doc):
+        match = re.search(r'<p.*?>(.*?)</p>', doc, re.DOTALL)
```

Comment:
It's not certain text is wrapped with `<p>` or that there's `<p>` in the first place. It's fairly common to use HTML when writing Javadocs, the tooling AFAIK doesn't automatically add `<p>` around the docs, and doc writers in general cannot be expected to add `<p>`s themselves.

Reply:
Non closing p are now covered